ckanext/s3filestore/plugin.py

```diff
@@ -78,16 +78,30 @@ def get_uploader(self, upload_to, old_filename=None):
     def after_update(self, context, pkg_dict):
         ''' Update the access of each S3 object to match the package.
         '''
-        LOG.debug("Package %s has been updated, notifying resources", pkg_dict['id'])
+        pkg_id = pkg_dict['id']
+        is_private = pkg_dict.get('private', False)
+        LOG.debug("after_update: Package %s has been updated, notifying resources", pkg_id)
+
+        # This is triggered repeatedly in the worker thread from plugins like
+        # 'validation' and 'archiver', so it needs to be efficient when
+        # no work is required.
+        latest_revision = toolkit.get_action('package_activity_list')(
+            context={'ignore_auth': True}, data_dict={'id': pkg_id, 'limit': 1})
```

Comment:
Does this have order by newest?

Reply:
yep, hidden in he limit function is the order by, https://github.com/ckan/ckan/blob/80941d1bd27557dbfa1351607fbec352361f1521/ckan/model/activity.py#L107

Reply:
I didn't actually delve into the code, but I knew that this was the same function used by the activity stream pages to display the most recent events, so it would definitely be ordered.